File: 04 - ORIENTAÇÃO A OBJETOS/nodeJS - JavaScript_/Exercicio 7/Com Pacotes/Professor.js
var Pessoa = require('./Pessoa')

module.exports = class Professor extends Pessoa {
    constructor(nome, endereco, idade, titulacao, listaDisc){
        super(nome, endereco, idade, listaDisc)
        this.titulacao = titulacao
    }

    printDescricao(){
        console.log(`Professor: ${this.nome}`)
        console.log(`Idade: ${this.idade}`)
        console.log(`Titulação: ${this.titulacao}`)
        const lista = this.listaDisc
        for(let i=0; i<lista.length; i++)
            console.log(`Nome: ${lista[i].nomeDisc} - Carga Horária: ${lista[i].cargaHoraria}`)
    }
}

File: 04-ORIENTACAO_A_OBJETOS/Java/Exercicio 8/Professor.java

import java.util.ArrayList;


public class Professor extends Pessoa {
    private String titulacao;

    public Professor(String nome, String endereco, int idade, String titulacao, ArrayList<Disciplina> listaDisc) {
        super(nome, endereco, idade, listaDisc);
        this.titulacao = titulacao;
    }
    
    String getTitulacao(){
        return this.titulacao;
    }

    @Override
    void printDescricao() {
        System.out.println("Professor: "+super.getNome());
        System.out.println("Endereço: "+super.getEndereco());
        System.out.println("Idade: "+super.getIdade());
        System.out.println("Titulação: "+this.titulacao);
        System.out.println("Disciplina(s) ministrada(s): ");
        ArrayList<Disciplina> listaDisc = super.getListaDisc();
        for(Disciplina disc : listaDisc){
            System.out.println("Nome: "+disc.getNomeDisc()+" - Carga Horária: "+disc.getCargaHoraria());
        }
    }
    
}
